checks/templates/mail-results.html

```diff
@@ -81,6 +81,18 @@ <h1>
     {% include "details-test-item.html" with testitem=details.dane_exists %}
     {% include "details-test-item.html" with testitem=details.dane_valid %}
     {% include "details-test-item.html" with testitem=details.dane_rollover %}
+
+  {% elif probe.name == 'mailrpki' %}
+    <div class="test-subsection">
+     {% include "string.html" with name="results domain-mail rpki mail-server label" %}
+    </div>
+    {% include "details-test-item.html" with testitem=details.mail_rpki_exists %}
+    {% include "details-test-item.html" with testitem=details.mail_rpki_valid %}
+    <div class="test-subsection">
+     {% include "string.html" with name="results domain-mail rpki name-servers label" %}
+    </div>
+    {% include "details-test-item.html" with testitem=details.ns_rpki_exists %}
+    {% include "details-test-item.html" with testitem=details.ns_rpki_valid %}
```

Comment:
I am open to splitting this out into its own subtest. I concur that this will improve clarity for the case where there are different nameservers for the MX records. I'm not sure if it's an improvement for the case where the sets are identical (that is, for most domains). I'll go ahead and just split the lists to see if it's an improvement.

This would also solve another issue in the /results_technical endpoint of the batch API, where the difference between the IPv6 test (not include MX NS) and the RPKI test (including the MX NS) results in some entries in the nameserver dict not having the reachable property, which I therefor had to turn optional. I'll go ahead and create a separate `mx-nameservers` dict mirroring the requested change.

Reply:
Note for future readers: following a conversation with @baknu, the implementation now returns the full list of nameservers for respetively the domain itself and the MX-records. For most domains, this will result in two identical technical_results lists. The reasoning is that the lists themselves become easier to understand.